packages/esm-patient-list-app/src/patientListList/index.tsx

```diff
@@ -0,0 +1,269 @@
+          onFocus={() => {
+            if (viewState.type === StateTypes.IDLE) setViewState({ type: StateTypes.SEARCH, searchTerm: '' });
+          }}
+          onBlur={() => {
+            if (
+              viewState.type === StateTypes.SEARCH ||
+              (viewState.type === StateTypes.SEARCH_WITH_RESULTS && viewState.searchTerm === '')
+            )
+              setViewState({ type: StateTypes.IDLE });
+          }}
+          onChange={({ target }) => {
+            if (target !== ref.current?.input) {
+              setTimeout(() => {
+                ref.current.input.blur();
+              }, 0);
+              setViewState({ type: StateTypes.IDLE });
+            } else {
+              setViewState((s) => ({ ...s, searchTerm: target.value }));
+            }
+          }}
+          onKeyPress={(e) => {
+            if (e.key === 'Enter')
+              setViewState(({ searchTerm }: SearchState) => ({
+                type: StateTypes.SEARCH_WITH_RESULTS,
+                searchTerm,
+                results: ['todo'],
+                enter: {},
+              }));
+          }}
+          ref={ref}
+          value={(viewState as SearchState)?.searchTerm || ''}
+        />
+      </div>
+      <div
+        style={{
+          gridRow: '2 / 3',
+          gridColumn: '1 / 2',
+          height: '100%',
+          width: '100%',
+          display: 'grid',
+          gridTemplate: '1.5rem auto / 1fr 1fr',
+          gap: '1rem',
+          padding: '1rem 1rem 0rem 1rem',
+          backgroundColor: 'white',
+        }}>
+        <p
+          style={{
+            height: '1.75rem',
+            fontSize: '1.25rem',
+            lineHeight: 1.4,
+          }}>
+          Patient Lists
+        </p>
+        <Button
+          style={{ width: 'fit-content', justifySelf: 'end', alignSelf: 'center' }}
+          kind="ghost"
+          renderIcon={Add16}
+          onClick={() => setRouteState({ type: RouteStateTypes.CREATE_NEW_LIST })}>
+          New List
+        </Button>
+        <Tabs
+          type="container"
+          style={{
+            gridColumn: 'span 2',
+          }}
+          tabContentClassName={'deactivate-tabs-content'}
+          onSelectionChange={setTabState}>
+          {(() => {
+            const res = [];
+            for (let index = 0; index < Object.keys(TabTypes).length / 2; index++) {
+              res.push(<Tab label={labelMap[index]} id={'tab-' + index}></Tab>);
+            }
+            return res;
+          })()}
+        </Tabs>
+      </div>
+      <div style={{ gridRow: '3 / 4', gridColumn: '1 / 2', height: '100%' }}>
+        <PatientListTable
+          loading={loading}
+          headers={customHeaders}
+          patientData={patientData}
+          setListStarred={setListStarred}
+          openPatientList={(listUuid) => {
+            setRouteState({ type: RouteStateTypes.SINGLE_LIST, listUuid });
+          }}
+        />
+      </div>
+      {(() => {
```

Comment:
I don't like these IIFE inside a React tree construction. Can't we use explicit components for that? Especially in this case an simple pattern would be more clear and flexible.

Reply:
I don't agree with your opinion on the IIFE but I moved the logic to a separate component.